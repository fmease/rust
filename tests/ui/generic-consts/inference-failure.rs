#![feature(generic_consts)]
#![allow(incomplete_features)]

const NONE<T>: Option<T> = None::<T>;

fn main() {
    let _ = NONE; //~ ERROR type annotations needed
}
