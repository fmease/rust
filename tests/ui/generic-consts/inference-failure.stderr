error[E0282]: type annotations needed for `Option<T>`
  --> $DIR/inference-failure.rs:7:9
   |
LL |     let _ = NONE;
   |         ^
   |
help: consider giving this pattern a type, where the type for type parameter `T` is specified
   |
LL |     let _: Option<T> = NONE;
   |          +++++++++++

error: aborting due to previous error

For more information about this error, try `rustc --explain E0282`.
